pkg/webhook/workspace/handler/immutable.go

```diff
@@ -0,0 +1,49 @@
+package handler
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/che-incubator/che-workspace-operator/pkg/apis/workspace/v1alpha1"
+	"github.com/che-incubator/che-workspace-operator/pkg/config"
+	"github.com/google/go-cmp/cmp"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+func (h *WebhookHandler) HandleImmutable(ctx context.Context, req admission.Request) admission.Response {
```

Comment:
I think `HandleImmutable` and `handleImmutableWorkspace` should be both private or public.

Reply:
The convention I was going for was:
- Function takes `(context, request)` and returns `admission.Response` -> public
- Function takes `old, new <object>` and returns (continue, message)` -> private

Perhaps the naming of the functions is not clear, but the goal is to have standalone methods that can be called from `mutate.go` be exported, and internal methods to be helpers for e.g. `handleImmutableWorkspace`.